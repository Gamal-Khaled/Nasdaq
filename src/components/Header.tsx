import React from "react";
import { Image, StyleSheet } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";

import theme from "../theme/theme";
import { spacing } from "../theme/spacing";

interface Props {
  handleSafeArea?: boolean;
}

const HeaderComponent = ({ handleSafeArea }: Props) => {
  return (
    <SafeAreaView
      edges={handleSafeArea ? ["top"] : []}
      style={styles.container}
    >
      <Image source={require("../assets/NASDAQ_Logo.png")} style={styles.logo} />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: spacing.base,
    paddingVertical: spacing.small,
    backgroundColor: theme.primary,
    flexDirection: "row",
  },
  logo: {
    width: 100,
    resizeMode: 'contain',
    tintColor: theme.font,
    aspectRatio: 3.5
  }
});

const Header = React.memo(HeaderComponent);
export default Header;
