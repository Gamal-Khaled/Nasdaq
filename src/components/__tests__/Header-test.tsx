import { cleanup, render, screen } from "@testing-library/react-native";

import Header from "../Header";

afterEach(cleanup);

describe("Testing 'Header' component", () => {
  test("Should render with default variant if no variant specified", () => {
    render(<Header />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
