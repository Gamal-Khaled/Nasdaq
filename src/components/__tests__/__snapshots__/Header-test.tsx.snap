// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing 'Header' component Should render with default variant if no variant specified 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "off",
      "left": "off",
      "right": "off",
      "top": "off",
    }
  }
  style={
    {
      "backgroundColor": "#191926",
      "flexDirection": "row",
      "paddingHorizontal": 16,
      "paddingVertical": 12,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/NASDAQ_Logo.png",
      }
    }
    style={
      {
        "aspectRatio": 3.5,
        "resizeMode": "contain",
        "tintColor": "#fff",
        "width": 100,
      }
    }
  />
</RNCSafeAreaView>
`;
